### PR TITLE
Metal BRDF changes

### PR DESCRIPTION
**Pull request by est77**
- Implemented fresnel conductor variant from "Fresnel Equations Considered Harmful".
- Use the energy compensation variant from "Practical multiple scattering compensation for microfacet models"
- Update OSL shaders.

### CODE REVIEW

File: src/appleseed/renderer/modeling/bsdf/metalbrdf.cpp [hunk 1]
```diff
@@ -299,25 +297,17 @@ namespace
         {
             if (values->m_energy_compensation != 0.0f)
             {
-                const float Ess =
-                    get_directional_albedo(
-                        std::abs(dot(outgoing, n)),
-                        values->m_roughness);
+                const float Ess = get_directional_albedo(
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bsdf/metalbrdf.cpp]: `ess` (lower case)

What does it stand for?

File: src/appleseed.shaders/src/appleseed/as_metal.osl [hunk 2]
```diff
@@ -62,6 +62,15 @@ shader as_metal
         string help = "Reflectance at grazing incidence.",
         int as_max_param_id = 2
     ]],
+    float in_edge_reflectance_weight = 0.0
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_metal.osl]: Shouldn't the default be 1?

Comment (est77) [reply src/appleseed.shaders/src/appleseed/as_metal.osl]: No. Should be 0 for compatibility with the metal roughness PBR workflow.

File: src/appleseed/renderer/modeling/bsdf/metalbrdf.cpp (hunk 1 above)
Comment (est77) [reply src/appleseed/renderer/modeling/bsdf/metalbrdf.cpp]: This comes from the original paper. E is for energy and ss for single scattering.

**Pull request merged**